### PR TITLE
Fix tests on Windows

### PR DESCRIPTION
**Pull request by Kha**
This reimplements 704e203.

### CODE REVIEW

File: library/Init/Data/String/Basic.lean [hunk 1]
```diff
@@ -155,20 +155,33 @@ private def utf8ExtractAux₁ : List Char → Pos → Pos → Pos → List Char
...
-    if sep.atEnd j then splitAux i i 0 (s.extract b (i-j)::r)
-    else splitAux b i j r
-  else splitAux b (s.next i) 0 r
+    if sep.atEnd j then splitOnAux i i 0 (s.extract b (i-j)::r)
+    else splitOnAux b i j r
+  else splitOnAux b (s.next i) 0 r
 
-def split (s : String) (sep : String := " ") : List String :=
-if sep == "" then [s] else splitAux s sep 0 0 0 []
+def splitOn (s : String) (sep : String := " ") : List String :=
```
Comment (Kha) [inline library/Init/Data/String/Basic.lean]: I followed the Haskell naming here

**Comment by Kha** [comment]
I'm suddenly a bit confused about https://github.com/leanprover/lean4/commit/704e203cd735e3361b29ae632f3f034dce1d07a6, did the old frontend not support absolute Windows paths (which contain `:`)? Do we want to?

**Comment by leodemoura** [comment]
> I'm suddenly a bit confused about [704e203](https://github.com/leanprover/lean4/commit/704e203cd735e3361b29ae632f3f034dce1d07a6), did the old frontend not support absolute Windows paths (which contain `:`)? Do we want to?

I am confused too. Not sure whether we want this feature or not in the new frontend. I want to cry whenever I have to work on these platform-dependent issues :)

**Comment by Kha** [comment]
Yeah, we can try to instead fix the use sites of LEAN_PATH to use `;` on Windows and come back to this patch if that becomes too bothersome.

**Comment by leodemoura** [comment]
merged

**Pull request closed without merging**